rust/automerge/src/transaction/transactable.rs

```diff
@@ -88,6 +89,21 @@ pub trait Transactable: ReadDoc {
         text: &str,
     ) -> Result<(), AutomergeError>;
 
+    fn mark<O: AsRef<ExId>>(
+        &mut self,
+        obj: O,
+        mark: Mark<'_>,
+        expand: (bool, bool),
```

Comment:
I think that rather than having these two booleans, which are hard to read for the caller, we should use an enum like:

```rust
enum ExpandMark  {
    Left,
    Right,
    Both,
    None,
}

impl ExpandMark {
    fn left(&self) -> bool {
        matches!(self, Self::Left | Self:Both)
    }
    fn right(&self) -> bool {
        matches!(self, Self::Right | Self::Both)
    }
}
```